hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/main/java/org/apache/hadoop/yarn/server/federation/store/records/ApplicationHomeSubCluster.java

```diff
@@ -150,7 +171,8 @@ public int hashCode() {
     return new HashCodeBuilder().
         append(this.getApplicationId()).
         append(this.getHomeSubCluster()).
-        append(this.getCreateTime()).toHashCode();
+        append(this.getCreateTime()).
+        append(this.getApplicationSubmissionContext()).toHashCode();
```

Comment:
Put the `toHashCode()` in the next line to avoid changes in future changes.

Reply:
I will fix it.